src/hooks/swap/useSwapInfo.tsx

```diff
@@ -45,71 +45,76 @@ interface SwapInfo {
   impact?: PriceImpact
 }
 
-// from the current swap inputs, compute the best trade and return it.
+/** Returns the best computed swap (trade/wrap). */
 function useComputeSwapInfo(routerUrl?: string): SwapInfo {
+  const { type, amount, [Field.INPUT]: inputCurrency, [Field.OUTPUT]: outputCurrency } = useAtomValue(swapAtom)
   const { account, chainId, isActivating, isActive } = useWeb3React()
   const isSupported = useOnSupportedNetwork()
-  const { type, amount, [Field.INPUT]: currencyIn, [Field.OUTPUT]: currencyOut } = useAtomValue(swapAtom)
   const isWrap = useIsWrap()
 
-  const chainIn = currencyIn?.chainId
-  const chainOut = currencyOut?.chainId
-  const tokenChainId = chainIn || chainOut
+  const inputChainId = inputCurrency?.chainId
+  const outputChainId = outputCurrency?.chainId
+  const tokenChainId = inputChainId || outputChainId
   const error = useMemo(() => {
     if (!isActive) return isActivating ? ChainError.ACTIVATING_CHAIN : ChainError.UNCONNECTED_CHAIN
     if (!isSupported) return ChainError.UNSUPPORTED_CHAIN
-    if (chainIn && chainOut && chainIn !== chainOut) return ChainError.MISMATCHED_TOKEN_CHAINS
+    if (inputChainId && outputChainId && inputChainId !== outputChainId) return ChainError.MISMATCHED_TOKEN_CHAINS
     if (chainId && tokenChainId && chainId !== tokenChainId) return ChainError.MISMATCHED_CHAINS
     return
-  }, [chainId, chainIn, chainOut, isActivating, isActive, isSupported, tokenChainId])
+  }, [chainId, inputChainId, outputChainId, isActivating, isActive, isSupported, tokenChainId])
 
   const parsedAmount = useMemo(
-    () => tryParseCurrencyAmount(amount, (isExactInput(type) ? currencyIn : currencyOut) ?? undefined),
-    [amount, type, currencyIn, currencyOut]
+    () => tryParseCurrencyAmount(amount, (isExactInput(type) ? inputCurrency : outputCurrency) ?? undefined),
+    [amount, type, inputCurrency, outputCurrency]
   )
-  const hasAmounts = currencyIn && currencyOut && parsedAmount && !isWrap
+  const hasAmounts = inputCurrency && outputCurrency && parsedAmount && !isWrap
   const trade = useRouterTrade(
     type,
     hasAmounts ? parsedAmount : undefined,
-    hasAmounts ? (isExactInput(type) ? currencyOut : currencyIn) : undefined,
-    RouterPreference.TRADE,
+    hasAmounts ? (isExactInput(type) ? outputCurrency : inputCurrency) : undefined,
+    isWrap || error ? RouterPreference.SKIP : RouterPreference.TRADE,
     routerUrl
   )
 
-  const amountIn = useMemo(
-    () => (isWrap || isExactInput(type) ? parsedAmount : trade.trade?.inputAmount),
-    [isWrap, parsedAmount, trade.trade?.inputAmount, type]
-  )
-  const amountOut = useMemo(
-    () => (isWrap || !isExactInput(type) ? parsedAmount : trade.trade?.outputAmount),
-    [isWrap, parsedAmount, trade.trade?.outputAmount, type]
-  )
```

Comment:
strong preference to keep these variable names: https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L587-L590